src/components/dialogs/input-hooks.js

```diff
@@ -424,74 +354,99 @@ export const useConnectivityValue = ({
     return [connectivity, render];
 };
 
+const filter = createFilterOptions();
+
 export const useAutocompleteField = ({
+    id,
     label,
     validation = {},
     inputForm,
     formProps,
     values,
+    getLabel = func_identity,
+    allowNewValue = false,
+    errorMsg,
 }) => {
     const [value, setValue] = useState('');
-
-    const intl = useIntl();
+    const [error, setError] = useState('');
+    const validationRef = useRef();
+    validationRef.current = validation;
 
     useEffect(() => {
         function validate() {
-            return true;
+            const res = validateField('' + value, validationRef.current);
+            setError(res?.errorMsgId);
+            return !res.error;
         }
-        inputForm.addValidation(label, validate);
-    }, [label, validation, inputForm, value]);
+        inputForm.addValidation(id ? id : label, validate);
+    }, [label, validation, inputForm, value, id]);
 
     const handleChangeValue = useCallback((value) => {
         setValue(value);
     }, []);
 
     const field = useMemo(() => {
         return (
-            // <Grid item xs={4} align="start">
             <Autocomplete
                 id={label}
                 onChange={(event, newValue) => {
                     handleChangeValue(newValue);
                 }}
-                options={Object.keys(values.object())}
-                getOptionLabel={(code) => values.get(code)}
+                size={'small'}
+                options={values}
+                getOptionLabel={getLabel}
+                {...(allowNewValue && {
+                    freeSolo: true,
+                    filterOptions: (options, params) => {
+                        const filtered = filter(options, params);
+                        if (
+                            params.inputValue !== '' &&
+                            !options.find((opt) => opt.id === params.inputValue)
+                        ) {
+                            filtered.push({
+                                inputValue: params.inputValue,
+                                id: params.inputValue,
+                            });
+                        }
+                        return filtered;
+                    },
+                })}
                 renderInput={(props) => (
                     <TextField
-                        {...formProps}
-                        {...props}
                         variant="filled"
```

Comment:
Removed. We get it from formProps

Reply:
change reverted

Reply:
Yes but variant="filled" has to be removed